src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -62,7 +61,7 @@ public HubConnection(IConnection connection, IHubProtocol protocol, ILoggerFacto
             _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
             _logger = _loggerFactory.CreateLogger<HubConnection>();
             _connection.OnReceived((data, state) => ((HubConnection)state).OnDataReceivedAsync(data), this);
-            _connection.Closed += Shutdown;
+            _connection.Closed.ContinueWith((task) => Shutdown(task.Exception));
```

Comment:
This is interesting. Users that subscribe to the Closed callback will add a continuation to connection.Closed task. I'm not sure continuations for tasks run sequentially. It might be better to expose a separate task on HubConnection that runs after Shutdown runs instead of doing it like this.